File: Raven.Abstractions/Data/Alert.cs
using System;

namespace Raven.Abstractions.Data
{
	public class Alert
	{
		public string Title { get; set; }
		public DateTime CreatedAt { get; set; }
		public bool Observed { get; set; }
        /// <summary>
        /// Purpose of this field is to avoid user from being flooded by recurring errors. We can display error i.e. once per day.
        /// This field might be used to determinate when user dismissed given alert for the last time.
        /// </summary>
        public DateTime? LastDismissedAt { get; set; }
		public string Message { get; set; }
		public AlertLevel AlertLevel { get; set; }
		public string Exception { get; set; }

		public string UniqueKey { get; set; }
	}

	public enum AlertLevel
	{
		Warning,
		Error
	}
}

File: Raven.Studio.Html5/App/models/alert.ts
class alert {
    alertLevel: string;
    createdAt: string;
    exception: string;
    message: string;
    title: string;
    uniqueKey: string;
    observed = ko.observable(false);

    isVisible: KnockoutComputed<boolean>;
    createdAtHumanized: KnockoutComputed<string>;

    constructor(dto: alertDto) {
        this.alertLevel = dto.AlertLevel;
        this.createdAt = dto.CreatedAt;
        this.exception = dto.Exception;
        this.message = dto.Message;
        this.observed(dto.Observed);
        this.title = dto.Title;
        this.uniqueKey = dto.UniqueKey;
    }

    toDto(): alertDto {
        return {
            AlertLevel: this.alertLevel,
            CreatedAt: this.createdAt,
            Exception: this.exception,
            Message: this.message,
            Observed: this.observed(),
            Title: this.title,
            UniqueKey: this.uniqueKey
        };
    }
}

export = alert;